cupy/core/core.pyx

```diff
@@ -744,49 +744,68 @@ cdef class ndarray:
 
         thrust.sort(self.dtype, self.data.ptr, self._shape)
 
-    def argsort(self):
-        """Return the indices that would sort an array with stable sorting
+    def argsort(self, axis=-1):
+        """Returns the indices that would sort an array with stable sorting
 
-        .. note::
-            For its implementation reason, ``ndarray.argsort`` currently
-            supports only arrays with their rank of one, and does not support
-            ``axis``, ``kind`` and ``order`` parameters that
-            ``numpy.ndarray.argsort`` supports.
+        Args:
+            axis (int or None): Axis along which to sort. Default is -1, which
+                means sort along the last axis. If None is supplied, the array
+                is flattened before sorting.
+
+        Returns:
+            cupy.ndarray: Array of indices that sort the array.
 
         .. seealso::
             :func:`cupy.argsort` for full documentation,
             :meth:`numpy.ndarray.argsort`
 
         """
 
-        # TODO(takagi): Support axis argument.
         # TODO(takagi): Support kind argument.
 
+        cdef Py_ssize_t ndim = self.ndim
+
         if not cupy.cuda.thrust_enabled:
             raise RuntimeError('Thrust is needed to use cupy.argsort. Please '
                                'install CUDA Toolkit with Thrust then '
                                'reinstall CuPy after uninstalling it.')
 
-        if self.ndim == 0:
+        if ndim == 0:
             raise ValueError('Sorting arrays with the rank of zero is not '
                              'supported')  # as numpy.argsort() raises
 
-        # TODO(takagi): Support ranks of two or more
-        if self.ndim > 1:
-            raise NotImplementedError('Sorting arrays with the rank of two or '
-                                      'more is not supported')
-
-        data = self.copy()
+        if axis is None:
+            data = self.reshape(self.size)
+            axis = -1
+        else:
+            data = self
 
-        # Assuming that Py_ssize_t can be represented with numpy.int64.
-        assert cython.sizeof(Py_ssize_t) == 8
+        if axis < 0:
+            axis += ndim
+        if not (0 <= axis < ndim):
+            raise ValueError('Axis out of range')
 
-        idx_array = ndarray(self.shape, dtype=numpy.int64)
+        if axis == ndim - 1:
+            data = data.copy(order='C')
+        else:
+            data = cupy.ascontiguousarray(cupy.rollaxis(data, axis, ndim))
```

Comment:
Please use `copy`. For example, if input shape is `(1, 1, 1)`, array will not be copy.